aten/src/ATen/core/qualified_name.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <c10/util/intrusive_ptr.h>
+#include <string>
+
+namespace c10 {
+
+// Represents names of the form, e.g., self.a.b
+struct QualifiedName;
+using QualifiedNamePtr = c10::intrusive_ptr<QualifiedName>;
+struct QualifiedName : c10::intrusive_ptr_target {
+  QualifiedName(QualifiedNamePtr prefix, std::string name)
+      : prefix_(std::move(prefix)), name_(std::move(name)) {
+    const auto pos = name_.find('.');
+    AT_ASSERTM(
```

Comment:
Is there a valid reason to have an empty string name?

Reply:
There needs to be a null state if you want to, e.g. default construct QualifiedNames to put them in maps.